### PR TITLE
Profiler improvement

### PR DESCRIPTION
**Pull request by 4z0t**
Profiler UI improvement.
Profiler done with main window and scroll area highlighting rows on hover.

### CODE REVIEW

File: lua/ui/game/Profiler.lua [hunk 1]
```diff
@@ -141,10 +149,18 @@ function CreateWindow()
...
+        end)
+        parent.OverviewTab.Left:Set(function()
+            return parent.Left()
+        end)
+        parent.OverviewTab.Width:Set(function()
+            return window.Width()
+        end)
+        parent.OverviewTab.Height:Set(function()
+            return 10
+        end)
```
Comment (HUSSAR-mtrela) [inline lua/ui/game/Profiler.lua:163]: Moving such short code to be on a new line, goes against FA programming style where authors made an effort to keep code easier to read and shorter by not adding new lines in dynamic functions.  Please can you undo adding new lines and keep your other changes?

**Review by HUSSAR-mtrela — CHANGES_REQUESTED** [review]

File: lua/ui/game/Profiler.lua [hunk 2]
```diff
@@ -522,7 +584,230 @@ function CloseWindow()
...
 
-    if State.GUI then 
+    if State.GUI then
         State.GUI:Hide()
     end
-end
\ No newline at end of file
+end
+
+ProfilerWindow = Class(Window.Window) {
```
Comment (HUSSAR-mtrela) [inline lua/ui/game/Profiler.lua:592]: please move ProfilerWindow class to a new file, e.g. ProfilerWindow.lua

File: lua/ui/game/Profiler.lua (hunk 1 above)
Comment (4z0t) [reply lua/ui/game/Profiler.lua:163]: I haven't completed everything yet, so yeah, i will do. Also, it does formatter, not me ☺️

File: lua/ui/game/ProfilerElements.lua [hunk 3]
```diff
@@ -132,15 +192,21 @@ function CreateScrollableContent(area, create, populate, empty)
...
                 populate(self.UIElements[k], self.Elements[index])
-            else 
+            else
                 empty(self.UIElements[k])
             end
         end
     end
 end
+
+ScrollArea = Class(Group) {
```
Comment (HUSSAR-mtrela) [inline lua/ui/game/ProfilerElements.lua:207]: the ScrollArea class is not used anywhere in this PR. Did you forget to use it or is still work-in-progress? If you want to keep it then you should move it to a new file under:
/lua/ui/controls/ScrollArea.lua

**Review by HUSSAR-mtrela — CHANGES_REQUESTED** [review]

File: lua/ui/game/Profiler.lua (hunk 2 above)
Comment (Garanas) [reply lua/ui/game/Profiler.lua:592]: Please to keep the `Profiler` keyword somewhere in there, it makes it easier to find all the profiler related files.

@HUSSAR-mtrela I've messaged you on Zulip - I don't know how to send messages on Github 😄

**Comment by Garanas** [comment]
10/10, will merge twice

**Comment by Garanas** [comment]
@4z0t Do you intend to make other changes? Otherwise I'll review the pull request and try to merge later tonight or tomorrow.

**Comment by 4z0t** [comment]
@Garanas I would like to do other tabs at least define methods for switching between them.

**Comment by Garanas** [comment]
I'll finalize the information to display for the benchmarks tab and the toggles for the options tab.

**Pull request merged**